### PR TITLE
Hotfix/target data issues

### PR DESCRIPTION
**Pull request by tnoelcke**
# Changes introduced

This pr fixes issues with undefined data causing the app to crash when navigating to the target page.

## Related issue(s):
CSE-357

## Contributor checklist

- [ ] Wrote/updated tests for introduced changes
- [ ] Added new stories for created/modified components (if applicable)
- [ ] Checked Storybook for Accessibility issues (if applicable)
- [ ] ~~Updated Postman library for created/modified routes (if applicable)~~
- [x] Verified that there are no issues in CircleCI
- [ ] Assigned yourself to the PR
- [ ] Removed `WIP:` from the PR title
- [ ] Requested review from the SB Reviewers team

## Reviewer checklist

- [ ] Assigned yourself to the PR
- [ ] Read through the changes in the `Files changed` tab
- [ ] Verified that tests were written/modified
- [ ] Verified that stories were written/modified (if applicable)
- [ ] ~~Verified that Postman was updated (if applicable)~~
- [ ] Checked out the branch and tested changes locally
- [ ] Run storybook and checked that there are no accessibility issues

### CODE REVIEW

File: src/models/target/index.ts [hunk 1]
```diff
@@ -32,20 +32,20 @@ export interface ITarget {
   title: string;
   shortCode: string;
   description: string;
-  standards: IStandards[];
+  standards?: IStandards[];
```
Comment (danvanhorn) [inline src/models/target/index.ts]: We should use union types `standards: IStandards | undefined` for this and use `undefined` specifically. This says "the property has not been initialized" vs "the property is not available"

Comment (danvanhorn) [reply src/models/target/index.ts]: The inventor of `null` in javascript also says that you should use `undefined`

**Review by danvanhorn — CHANGES_REQUESTED** [review]
Lets go with union types in our Target interface

File: src/models/target/index.ts (hunk 1 above)
Comment (tnoelcke) [reply src/models/target/index.ts]: Ok, I'll do that and get it pushed up. I agree that is a good idea.

**Review by danvanhorn — APPROVED** [review]
This looks good.

**Pull request merged**